src/hera/workflows/v5/steps.py

```diff
@@ -0,0 +1,146 @@
+from typing import Any, Dict, List, Optional, Union
+
+from hera.workflows.models import (
+    Arguments as _ModelArguments,
+    Artifact as _ModelArtifact,
+    ContinueOn as _ModelContinueOn,
+    Item as _ModelItem,
+    LifecycleHook as _ModelLifecycleHook,
+    Parameter as _ModelParameter,
+    Sequence as _ModelSequence,
+    Template as _ModelTemplate,
+    TemplateRef as _ModelTemplateRef,
+    WorkflowStep as _ModelWorkflowStep,
+)
+from hera.workflows.v5._mixins import (
+    _ContextMixin,
+    _IOMixin,
+    _SubNodeMixin,
+    _TemplateMixin,
+)
+from hera.workflows.v5.exceptions import InvalidType
+
+
+class Step(
+    _SubNodeMixin,
+):
+    arguments: Optional[List[Union[_ModelArtifact, _ModelParameter]]] = None
+    continue_on: Optional[_ModelContinueOn]
+    hooks: Optional[Dict[str, _ModelLifecycleHook]]
+    inline: Optional[_ModelTemplate]
+    name: Optional[str]
+    on_exit: Optional[str]
+    template: Optional[str]
+    template_ref: Optional[_ModelTemplateRef]
+    when: Optional[str]
+    with_items: Optional[List[_ModelItem]]
+    with_param: Optional[str]
+    with_sequence: Optional[_ModelSequence]
+
+    def _build_as_workflow_step(self) -> _ModelWorkflowStep:
+        # Convert the unified list of Artifacts and Parameters in self.arguments to a
+        # model-class Arguments to pass to WorkflowStep
+        model_arguments = _ModelArguments(
+            artifacts=list(filter(lambda arg: isinstance(arg, _ModelArtifact), self.arguments or [])) or None,
+            parameters=list(filter(lambda arg: isinstance(arg, _ModelParameter), self.arguments or [])) or None,
+        )
+
+        return _ModelWorkflowStep(
+            arguments=model_arguments,
+            continue_on=self.continue_on,
+            hooks=self.hooks,
+            inline=self.inline,
+            name=self.name,
+            on_exit=self.on_exit,
+            template=self.template,
+            template_ref=self.template_ref,
+            when=self.when,
+            with_items=self.with_items,
+            with_param=self.with_param,
+            with_sequence=self.with_sequence,
+        )
+
+    def _build_as_parallel_steps(
+        self,
+    ) -> List[_ModelWorkflowStep]:
+        return [self._build_as_workflow_step()]
+
+
+class ParallelSteps(
+    _ContextMixin,
+    _SubNodeMixin,
+):
+    parallel_steps: List[Step] = []
+
+    def _add_sub(self, node: Any):
+        if not isinstance(node, Step):
+            raise InvalidType()
+        self.parallel_steps.append(node)
+
+    def _build_as_parallel_steps(self) -> List[_ModelWorkflowStep]:
+        return [step._build_as_workflow_step() for step in self.parallel_steps]
+
+
+class Steps(
+    _ContextMixin,
+    _IOMixin,
+    _SubNodeMixin,
+    _TemplateMixin,
+):
+    workflow_steps: List[Union[Step, ParallelSteps]] = []
```

Comment:
Can we also add the step model here?

Reply:
Is it actually worth adding the model? It would be unintuitive to users that they have to call `_add_sub` as the model WorkflowStep can't be a `_SubNodeMixin` - so maybe we should add an accompanying manual `add_step` function (which just forwards to `_add_sub`)?